nurseconnect/views.py

```diff
@@ -50,46 +62,266 @@ def search(request, results_per_page=10):
 
 
 class RegistrationView(FormView):
+    """
+    Handles user registration
+    """
     form_class = forms.RegistrationForm
     template_name = "profiles/register.html"
 
     def form_valid(self, form):
         username = form.cleaned_data["username"]
         password = form.cleaned_data["password"]
-        user = User.objects.create_user(username=username, password=password)
-        user.save()
-
-        authed_user = authenticate(username=username, password=password)
-        login(self.request, authed_user)
-        return HttpResponseRedirect(reverse("home"))
+        try:
```

Comment:
I can't say I see the purpose of this `try` block.  The `User.DoesNotExist` would never be raised.